packages/astro/src/runtime/client/dev-toolbar/toolbar.ts

```diff
@@ -553,6 +560,15 @@ export class AstroDevToolbar extends HTMLElement {
 			?.querySelector('#dropdown')
 			?.toggleAttribute('data-no-notification', !newStatus);
 	}
+
+	setToolbarPlacement(newPlacement: Placement) {
+		this.devToolbarContainer?.setAttribute('data-placement', newPlacement);
+		this.apps.forEach((app) => {
+			this.getAppCanvasById(app.id)
+				?.shadowRoot?.querySelector('astro-dev-toolbar-window')
+				?.setAttribute('data-placement', newPlacement);
+		});
```

Comment:
This should be done through a notification instead of hardcoding it for window, that way the apps can also update their other components if needed.

Reply:
By "notification" do you mean to dispatch custom events? I'm not sure I understand. Could you explain it a bit to me please? 🙏

Reply:
No worries! Yes, I mean by dispatching an event, sorry not sure why I said notification. Similar to how the `app-toggled` event work.

Reply:
Here's the adjusted version: https://github.com/withastro/astro/pull/10591/commits/a714cb10331b951b678769f2bdfca447dfeb6559